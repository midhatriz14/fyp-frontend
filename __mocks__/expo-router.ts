const mockRouter = {
  push: jest.fn(),
};

export const useRouter = () => mockRouter;