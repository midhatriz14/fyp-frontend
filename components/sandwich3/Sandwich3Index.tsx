import React from 'react';
import { View, StyleSheet, Image, Text, TouchableOpacity } from 'react-native';
import Header from './Header';
import { router } from 'expo-router';


interface EventPlannerScreenProps {}

const Sandwich3Index: React.FC<EventPlannerScreenProps> = () => {
  return (
    <View style={styles.container}>
      <Header />
      <Image
        resizeMode="contain"
        source={{ uri: "https://cdn.builder.io/api/v1/image/assets/TEMP/d24d1ebd4c9319e81764fb2995ce41f17aedb4023b3fa2a7a2c53b54641bae4e?placeholderIfAbsent=true&apiKey=0a92af3bc6e24da3a9ef8b1ae693931a" }}
        style={styles.mainImage}
      />
      <Text style={styles.title}>
        Enjoy Every Moment with Expert Event Planners!
      </Text>
      <Text style={styles.subtitle}>
        Let top event planners handle the details, so you can focus on enjoying your special day.
      </Text>
      <TouchableOpacity style={styles.getStartedButton}
         onPress={() => { router.push("/intro") }}>
        <Text style={styles.getStartedText}>Get Started</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 40,
    display: "flex",
    marginLeft: "auto",
    marginRight: "auto",
    maxWidth: 480,
    width: "100%",
    paddingBottom: 71,
    flexDirection: "column",
    overflow: "hidden",
    alignItems: "center",
    backgroundColor: '#F8E9F0'
  },
  mainImage: {
    position: "relative",
    display: "flex",
    marginTop: 9,
    width: "100%",
    maxWidth: 310,
    aspectRatio: 1.42,
  },
  title: {
    color: "rgba(0, 0, 0, 1)",
    fontSize: 16,
    fontFamily: "Poppins, sans-serif",
    fontWeight: "700",
    lineHeight: 16,
    letterSpacing: 0.8,
    textAlign: "center",
    textTransform: "uppercase",
    marginTop: 65,
  },
  subtitle: {
    color: "rgba(0, 0, 0, 1)",
    fontSize: 16,
    fontFamily: "Poppins, sans-serif",
    fontWeight: "400",
    lineHeight: 15,
    letterSpacing: 0.8,
    textAlign: "center",
    marginTop: 5,
  },
  getStartedButton: {
    alignSelf: "stretch",
    width: "100%",
    borderRadius: 10,
    marginTop: 75,
    maxWidth: 300,
    paddingLeft: 68,
    paddingRight: 68,
    paddingTop: 13,
    paddingBottom: 13,
    backgroundColor: "#780c61",
  },
  getStartedText: {
    fontFamily: "Poppins, sans-serif",
    fontSize: 16,
    color: "#FFF",
    fontWeight: "500",
    textAlign: "center",
  },
});

export default Sandwich3Index;