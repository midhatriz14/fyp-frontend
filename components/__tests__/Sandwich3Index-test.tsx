import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Sandwich3Index from "../sandwich3/Sandwich3Index";
import { router } from "expo-router";

jest.mock("expo-router", () => ({
  router: { push: jest.fn() }, // Mock router.push function
}));

describe("Sandwich3Index Screen", () => {
  test("renders all UI elements correctly", () => {
    const { getByText } = render(<Sandwich3Index />);

    // Title has nested text, so use partial matches with regex
    // Checking for "ENJOY EVERY MOMENT" and "EVENT PLANNERS!" portions:
    expect(getByText(/ENJOY EVERY MOMENT/)).toBeTruthy();
    expect(getByText(/EVENT PLANNERS!/)).toBeTruthy();

    // Check Description
    // "Let top event planners handle the details, so you can focus on enjoying your special day."
    // The provided text is:
    // "Let top event planners handle the details, so you can focus on enjoying your special day."
    // We'll match a unique part of this sentence to ensure itâ€™s rendered.
    expect(getByText(/Let top event planners handle the details/)).toBeTruthy();

    // Check the "Get Started" button text
    expect(getByText("Get Started")).toBeTruthy();
  });

  test("navigates to /intro when Get Started is pressed", () => {
    const { getByText } = render(<Sandwich3Index />);
    const getStartedButton = getByText("Get Started");
    fireEvent.press(getStartedButton);
    expect(router.push).toHaveBeenCalledWith("/intro");
  });

  // Optional: Snapshot test to ensure UI consistency
  test("matches the snapshot", () => {
    const { toJSON } = render(<Sandwich3Index />);
    expect(toJSON()).toMatchSnapshot();
  });

  test("does not render irrelevant text", () => {
    const { queryByText } = render(<Sandwich3Index />);
    expect(queryByText("This text should not be here")).toBeNull();
  });
  test("checks highlight text presence", () => {
    const { getByText } = render(<Sandwich3Index />);
    // Check a unique word in highlight text
    expect(getByText(/EVENT PLANNERS!/)).toBeTruthy();
  });
  test("description contains specific phrases", () => {
    const { getByText } = render(<Sandwich3Index />);
    expect(getByText(/Let top event planners handle the details/)).toBeTruthy();
    expect(
      getByText(/so you can focus on enjoying your special day/)
    ).toBeTruthy();
  });
  test("button text is correct", () => {
    const { getByText } = render(<Sandwich3Index />);
    expect(getByText(/Get Started/)).toBeTruthy();
  });
    test("UI does not change unexpectedly", () => {
      const { toJSON } = render(<Sandwich3Index />);
      expect(toJSON()).toMatchSnapshot();
    });
  test("no other route is called on button press", () => {
    const { getByText } = render(<Sandwich3Index />);
    const getStartedButton = getByText("Get Started");
    fireEvent.press(getStartedButton);
    // Expect correct route
    expect(router.push).toHaveBeenCalledWith("/intro");
    // Check koi aur route call na hua
    expect(router.push).not.toHaveBeenCalledWith("/sandwich2");
  });
});
