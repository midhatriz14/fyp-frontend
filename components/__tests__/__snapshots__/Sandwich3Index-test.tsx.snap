// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sandwich3Index Screen UI does not change unexpectedly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F8E9F0",
      "flex": 1,
      "justifyContent": "center",
      "paddingVertical": 40,
    }
  }
>
  <Image
    source={
      {
        "uri": undefined,
      }
    }
    style={
      {
        "height": "40%",
        "resizeMode": "contain",
        "width": "80%",
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "marginTop": 20,
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      "ENJOY EVERY MOMENT WITH EXPERT 
      <Text
        style={
          {
            "color": "#E15A45",
          }
        }
      >
        EVENT PLANNERS!
      </Text>
      "
    </Text>
    <Text
      style={
        {
          "color": "#555",
          "fontSize": 16,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      Let top event planners handle the details, so you can focus on enjoying your special day.
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "bottom": 40,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "left": 0,
        "paddingHorizontal": 30,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#780C60",
          "borderRadius": 10,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 40,
          "paddingVertical": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "600",
          }
        }
      >
        Get Started
      </Text>
    </View>
  </View>
</View>
`;

exports[`Sandwich3Index Screen matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F8E9F0",
      "flex": 1,
      "justifyContent": "center",
      "paddingVertical": 40,
    }
  }
>
  <Image
    source={
      {
        "uri": undefined,
      }
    }
    style={
      {
        "height": "40%",
        "resizeMode": "contain",
        "width": "80%",
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "marginTop": 20,
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      "ENJOY EVERY MOMENT WITH EXPERT 
      <Text
        style={
          {
            "color": "#E15A45",
          }
        }
      >
        EVENT PLANNERS!
      </Text>
      "
    </Text>
    <Text
      style={
        {
          "color": "#555",
          "fontSize": 16,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      Let top event planners handle the details, so you can focus on enjoying your special day.
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "bottom": 40,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "left": 0,
        "paddingHorizontal": 30,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#780C60",
          "borderRadius": 10,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 40,
          "paddingVertical": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "600",
          }
        }
      >
        Get Started
      </Text>
    </View>
  </View>
</View>
`;
